grc/gui/ParamWidgets.py

```diff
@@ -334,3 +334,47 @@ def _handle_clicked(self, widget=None):
             self._editing_callback()
             self._apply_change()
         file_dialog.destroy()  # destroy the dialog
+
+class DirectoryParam(EntryParam):
+    """Provide an entry box for a directory and a button to browse for it."""
+
+    def __init__(self, *args, **kwargs):
+        EntryParam.__init__(self, *args, **kwargs)
+        self._open_button = Gtk.Button(label='...')
+        self._open_button.connect('clicked', self._handle_clicked)
+        self.pack_start(self._open_button, False, False, 0)
```

Comment:
How do you feel about deriving from FileParam to avoid duplication the `__init__` here